internal/client/repository.go

```diff
@@ -874,6 +874,30 @@ func (bs *blobStatter) Stat(ctx context.Context, dgst digest.Digest) (v1.Descrip
 		return v1.Descriptor{}, err
 	}
 	defer resp.Body.Close()
+	// Some registries do not support HEAD requests
+	if resp.StatusCode == http.StatusForbidden {
```

Comment:
I'm not convinced this is a good idea. In fact I feel we shouldnt be doing this i.e. sending `GET` when remote doesnt "support" `HEAD`.